### ISSUE TITLE
GradNorm: Gradient Normalization for Adaptive Loss Balancing in Deep Multitask Networks

### ISSUE THREAD

**Issue by howardyclo** [howardyclo/papernotes#64]
### Metadata
- Authors: Zhao Chen, Vijay Badrinarayanan, Chen-Yu Lee, Andrew Rabinovich
- Organization: Magic Leap, Inc. (The 1st author is now in Waymo)
- Conference: ICML 2018
- Paper: https://arxiv.org/abs/1711.02257

### TL;DR
- Multi-task deep learning models suffer from imbalanced losses (or gradient norms) during training.
- Propose a gradient normalization method, GradNorm, for balancing multi-task losses, resulting in improved accuracy and reduced overfitting compared to single-task.

### Connection to Curriculum Learning
-  Curriculum learning: Tasks are subsequently trained based on local rewards.
-  This work: Tasks are jointly trained based on global rewards (e.g., total loss).

### Methods
> TBD.

### Implementation Tricks
- Apply GradNorm to limited shared layers (e.g., only last shared layer) can save computational costs.
- Initial loss can also be initialized to theoretical initial loss (e.g.,  log(C) for CE loss across C classes), instead of initial loss produced by initial network (to prevent unstable initialization of network).

### Personal Thoughts
- What if there are different difficulty levels across tasks? Is balancing learning multiple tasks a reasonable approach (i.e., same convergence rate between easy task and hard task)? Or is it reasonable approach for tasks that may need to be learned in a curriculum fashion?

### Read More
- [Multi-Task Learning Using Uncertainty to Weigh Losses for Scene Geometry and Semantics](https://arxiv.org/abs/1705.07115) by Kendall et al. CVPR 2018. Mentioned in #42. They use a joint likelihood formulation to derive task weights based on the intrinsic uncertainty in each task.
- [Multi-Task Learning as Multi-Objective Optimization](https://papers.nips.cc/paper/7334-multi-task-learning-as-multi-objective-optimization.pdf) by Sener & Koltun. NIPS 2018.
